diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2b32832..7af56ac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,9 +35,10 @@ add_library(dpf STATIC src/dpf.c)
 target_include_directories(dpf PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
 target_link_libraries(dpf PUBLIC OpenMP::OpenMP_C)
 
-add_library(dcf STATIC src/dcf.c)
+add_library(dcf STATIC src/dcf.c src/prg/aes_mmo_ni.c src/group/bytes.c)
 target_include_directories(dcf PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
 target_link_libraries(dcf PUBLIC OpenMP::OpenMP_C)
+target_compile_options(dcf PRIVATE -msse2 -maes)
 
 add_library(cw_mac_bytes STATIC src/cw_mac_bytes.c)
 target_include_directories(cw_mac_bytes PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
